test/README.md

```diff
@@ -1,23 +1,133 @@
+
+## Adding Tests to CMake Builds
+
+Tests should be added to the CMake build using [/test/CMakeLists.txt](./CMakeLists.txt).  Within that file, there are several test targets that are already defined, using either the `add_automated_test()` or `add_automated_test_w_mock()` macro.  In particular:
+
+* `test_unit` for all unit tests
+* `test_integration` for all integration tests
+* `test_all` for all unit and integration tests
+
+These and other similar targets are configured (using either the `add_automated_test()` or `add_automated_test_w_mock()` macro) with the applicable `*.cpp` files that have the code for tests that should be included in the particular target.  
+
+Existing target configurations must be updated with the test file for any newly added tests, assuming the file is not already configured.  
+
+## Test Creation Rules of Thumb
+
+There are few additional rules of thumb to keep in mind when creating new tests.  These are not set in stone yet; rather, they seem like good practices, but feedback is still needed on how useful they are and if there are any situations where they are lacking.
+
+* [Separate test types in separate files](#separate-test-types-in-separate-files)
+* [Use analogous names](#use-analogous-names)
+* [Use analogous paths](#use-analogous-paths)
+* [Keep unit test assertions to a minimum](#keep-unit-test-assertions-to-a-minimum)
+
+#### Separate Test Types in Separate Files
+
+Keep unit tests, integration tests, etc., in separate files dedicated to that type of test.  This is helpful for several reasons, including isolating which overall group of tests get run.
+
+#### Use Analogous Names
+
+To help give clarity to what relates to what, use analogous names for associated test items.  This should be the source code item's name(\*), with either a standard prefix or suffix added.
+
+- files
+    - base: source code filename
+    - suffix: `_Test`/`Test` for unit tests or `_IT`/`IT` for integration tests
+        - e.g., `LinearReservoir.cpp` and `LinearReservoirTest.cpp`
+        - e.g., `HY_FlowPath.cpp` and `HY_FlowPath_Test.cpp`
+- *test fixtures*, *test suites*, and associated classes
+    - base: *upper camel case* form(\*) of source code class
+    - suffix: `Test` for unit tests or `IT` for integration tests
+- test names
+    - base: *upper camel case* form(\*) of source code function
+    - prefix: `Test`
+    - suffix: optional(\*\*) alphanumeric identifier (e.g., `1`, `1a`, `1b`)
 
-The submodule is checked out to a particular tagged commit of **Google Test**. Over time, as new releases for **Google Test** are available, the tagged version used by this project may change.  Clones of this repo can stay in sync with any such submodule changes by re-running the `git submodule update --init --recursive` command.
+\* Underscores are not allowed in **test fixtures** by **Google Test**, thus the bases for these are belong to camel case.
```

Comment:
I'll make a change, but that's correct: test classes, which Google calls **test fixtures**, cannot be named using underscores.  Since the base for such a name may need to be changed, it makes sense to adopt a standard converting style and apply it to all cases.

The same is actually true for test names (within the scope of a suite), which this suggests basing on converted function names, so I'll add something for that too.

As for the wording, it appears my *Zero Wing* reference was poorly made.

Reply:
A change for this is now in the PR branch.